api/query/cache/index/filter.go

```diff
@@ -144,7 +171,7 @@ func newFilter(idx index, q query.ConcreteQuery) (filter, error) {
 		}
 		return Not{idx, f}, nil
 	default:
-		return nil, errUnknownConcreteQuery
+		return nil, fmt.Errorf("Unknown ConcreteQuery type %s", reflect.TypeOf(q))
```

Comment:
Hmm... I'd like to avoid using reflection if possible, especially for an informative error message.

Reply:
This line should basically never get hit, and when it does, debugging the type is quite important. This is how I noticed that we weren't translating True/False across filters.
We're an open source project; revealing the concrete type of a json.Unmarshal and bind isn't a security problem.